lib/modules/manager/custom/jsonata/utils.ts

```diff
@@ -0,0 +1,102 @@
+import { URL } from 'url';
+import is from '@sindresorhus/is';
+import jsonata from 'jsonata';
+import { logger } from '../../../../logger';
+import * as template from '../../../../util/template';
+import type { PackageDependency } from '../../types';
+import type { JSONataManagerTemplates, JsonataExtractConfig } from './types';
+
+export const validMatchFields = [
+  'depName',
+  'packageName',
+  'currentValue',
+  'currentDigest',
+  'datasource',
+  'versioning',
+  'extractVersion',
+  'registryUrl',
+  'depType',
+] as const;
+
+type ValidMatchFields = (typeof validMatchFields)[number];
+
+export async function handleMatching(
+  json: unknown,
+  packageFile: string,
+  config: JsonataExtractConfig,
+): Promise<PackageDependency[]> {
+  // Pre-compile all JSONata expressions once
+  const compiledExpressions = config.matchStrings
+    .map((query) => {
+      try {
+        return jsonata(query);
+      } catch (err) {
+        logger.warn(
+          { err },
+          `Failed to compile JSONata query: ${query}. Excluding it from queries.`,
+        );
+        return null;
+      }
+    })
+    .filter((expr) => expr !== null);
+
+  // Execute all expressions in parallel
+  const results = await Promise.all(
+    compiledExpressions.map(async (expr) => {
+      const result = (await expr.evaluate(json)) ?? [];
+      return is.array(result) ? result : [result];
+    }),
+  );
+
+  // Flatten results and create dependencies
+  return results
+    .flat()
+    .map((queryResult) => {
+      return createDependency(queryResult as Record<string, string>, config);
+    })
+    .filter((dep) => dep !== null);
+}
+
+export function createDependency(
+  queryResult: Record<string, string>,
+  config: JsonataExtractConfig,
+): PackageDependency | null {
+  const dependency: PackageDependency = {};
+
+  function updateDependency(field: ValidMatchFields, value: string): void {
+    switch (field) {
+      case 'registryUrl':
+        // check if URL is valid and pack inside an array
+        try {
+          const url = new URL(value).toString();
+          dependency.registryUrls = [url];
+        } catch {
+          logger.warn({ value }, 'Invalid json manager registryUrl');
```

Comment:
```suggestion
          logger.warn({ value }, 'Invalid JSONata manager registryUrl');
```
This is a good example why we should look into allowing people to name custom managers again. 

If somebody is getting this warning in a bigger setup, the user will have a hard time to identify the correct manager especially with presets.